src/cbc_sdk/platform/alerts.py

```diff
@@ -84,47 +84,48 @@ class Alert(PlatformModel):
     }
 
     REMAPPED_ALERTS_V7_TO_V6 = {
```

Comment:
In the to_json() method in the "if version == "v6" section where this is called around line 470, the "if key ==threat_cause_event_id" check is checking for the v6 field name when it is iterating over the v7 fields.

Instead of:
```
if key == "threat_cause_event_id":
                    modified_json["created_by_event_id"] = value
```
 
I think it should be:
              ```
  if key == "primary_event_id": # the v7 field name
                    # created_by_event_id set by REMAPPED_ALERTS_V7_TO_V6
                    # secondary mapping when alert type is CB_ANALYTICS needed to created_by_event_id
                    modified_json["threat_cause_cause_event_id"] = value # the other v6 field name that also holds the primary event id value
```